js/markDone.js

```diff
@@ -0,0 +1,23 @@
+import { EDIT, COMPLETE } from "/js/constants.js";
+import calculateDuration from "/js/utility/mDDuration.js";
+import createDurationElement from "/js/utility/mDDuElement.js";
+
+export default function markTodoAsCompleted(element, taskList, id) {
+  const index = taskList.findIndex((item) => item.id === id);
```

Comment:
Let's handle what'll happen when `findIndex` can't find the index.

Reply:
The findIndex() method returns the index of the first element in taskList that satisfies the provided (item) => item.id === id) function. If no elements satisfy the testing function, -1 is returned. returning -1 means taskList doesn't contain the task which will only happen if we do any logical error i.e can't update taskList according to requirements. To be in a safe zone I just prohibited unintended out-of-index access to taskList. Please check.